src/services/cronJobs/syncUserPoewrs.ts

```diff
@@ -62,41 +64,46 @@ async function addSyncUserPowerJobsToQueue() {
   const toTimeStamp =
     previousGivbackRound * givbackRoundLength + firstGivbackRoundTimeStamp;
 
-  const users = await findUsersThatDidntSyncTheirPower(previousGivbackRound);
-  users.forEach(user => {
+  let totalFetched = 0;
+
+  while (true) {
+    const [users, count] = await findUsersThatDidntSyncTheirPower(
+      previousGivbackRound,
+      totalFetched,
+    );
+
     syncUserPowersQueue.add({
-      userId: user.id,
+      users,
       fromTimeStamp,
       toTimeStamp,
       givbackRound: previousGivbackRound,
     });
-  });
+
+    totalFetched += users.length;
+    if (totalFetched >= count) break;
+  }
```

Comment:
I don't see this as an infinite loop, as the condition is clearly stated, could this be a good candidate for a do...while loop?
Something like
```
let totalFetched = 0;
let count = 0;
do {
 // logic and http calls
} while (totalFetched < count)
```

I ask because, (do...while) are not common. But seems like the code needs to execute at least once.

Reply:
I replaced it.
It's a matter of taste though. Some modern languages like Golang and Rust don't have it anymore.